src/globus_sdk/services/auth/client/service_client.py

```diff
@@ -631,3 +680,1187 @@ def delete_project(self, project_id: UUIDLike) -> GlobusHTTPResponse:
+    #   - domain_constraints_include=[] != domain_constriants_include=None
+    # - update_client():
+    #   - terms_and_conditions=None
+    #   - privacy_policy=None
+    #   - required_idp=None
+    #   - preselect_idp=None
+    class _DefaultNull:
+        ...
+
+    _DEFAULT_NULL = _DefaultNull()
+
+    def update_policy(
+        self,
+        policy_id: UUIDLike,
+        *,
+        project_id: UUIDLike | None = None,
+        authentication_assurance_timeout: int | None = None,
+        display_name: str | None = None,
+        description: str | None = None,
+        # TODO: include=null is not the same as include=[]
+        domain_constraints_include: t.Iterable[str]
+        | None
+        | _DefaultNull = _DEFAULT_NULL,
+        domain_constraints_exclude: t.Iterable[str]
+        | None
+        | _DefaultNull = _DEFAULT_NULL,
+    ) -> GlobusHTTPResponse:
+        """
+        Update a policy. Requires the ``manage_projects`` scope.
+
+        :param policy_id: ID of the policy to update
+        :type policy_id: str or uuid
+
+        :param project_id: ID of the project for the new policy
+        :type project_id: str or uuid
+        :param authentication_assurance_timeout: Number of seconds within which someone
+            must have authenticated to satisfy the policy
+        :type authentication_assurance_timeout: int
+        :param display_name: A user-friendly name for the policy
+        :type display_name: str
+        :param description: A user-friendly description to explain the purpose of the
+            policy
+        :type description: str
+        :param domain_constraints_include: A list of domains that can satisfy the policy
+        :type domain_constraints_include: iterable of str
+        :param domain_constraints_exclude: A list of domains that can not satisfy the
+            policy
+        :type domain_constraints_exclude: iterable of str
+
+        .. tab-set::
+
+            .. tab-item:: Example Usage
+
+                .. code-block:: pycon
+
+                    >>> ac = globus_sdk.AuthClient(...)
+                    >>> policy_id = ...
+                    >>> r = ac.update_policy(scope_id, display_name="Greenlight Policy")
+
+            .. tab-item:: Example Response Data
+
+                .. expandtestfixture:: auth.update_policy
+
+            .. tab-item:: API Info
+
+                ``POST /v2/api/policies/{policy_id}``
+
+                .. extdoclink:: Update Policy
+                    :ref: auth/reference/#update_policy
+        """
+        body: dict[str, t.Any] = {}
+
+        if project_id is not None:
+            body["project_id"] = str(project_id)
+        if authentication_assurance_timeout is not None:
+            body["authentication_assurance_timeout"] = authentication_assurance_timeout
+        if display_name is not None:
+            body["display_name"] = display_name
+        if description is not None:
+            body["description"] = description
```

Comment:
This is a good spot to call out -- there's a new way for you to use `MISSING` once you pick it up from `main`, and not just for the fields below with `_DEFAULT_NULL`. You can use it on these default-`None` fields too!

```python
from globus_sdk.utils import MissingType, MISSING

...

    def update_policy(
        ...
        authentication_assurance_timeout: int | MissingType = MISSING,
        display_name: str | MissingType = MISSING,
        description: str | MissingType = MISSING,
        ...
    ) -> GlobusHTTPResponse:
        ...
        body["authentication_assurance_timeout"] = authentication_assurance_timeout
        body["display_name"] = display_name
        body["description"] = description
```

_drops the mic_

I think this is easier on all of us to write and read, and the `MISSING` values will get auto-stripped out of the request.

---

I haven't decided if I think it's safe for us to go back and make a lot of `None`-type things default to `MISSING` instead. I think it might be a mild enough compatibility break in a lot of cases that it will be worth it, but maybe it's better to save that all for v4.